### PR TITLE
mysql: make ComQuery streamable

### PR DESCRIPTION
**Pull request by sougou**
ComQuery signature has been changed to use a callback, which
can be used for streaming results.

The function the handles ComQuery became slightly complex for
two reasons:
1. If there's no field info (DMLs), no further packs must be sent.
2. There is no way to send an error mid-stream (I tested it). All
   we can do is abort, which closes the connection.

### CODE REVIEW

File: go/vt/vtgate/plugin_mysql_server.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func (vh *vtgateHandler) ConnectionClosed(c *mysql.Conn) {
 	}
 }
 
-func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query []byte) (*sqltypes.Result, error) {
+func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query []byte, callback func(*sqltypes.Result) error) error {
 	// FIXME(alainjobart): Add some kind of timeout to the context.
```
Comment (alainjobart) [inline go/vt/vtgate/plugin_mysql_server.go]: At some point we have to think about that one, but not part of this PR, obviously.

**Comment by sougou** [comment]
Tagging #2933
